File: sb-web/src/main/java/org/example/model/Result.java
package org.example.model;

public class Result<T>{
    public int code = 0;
    public String msg = "成功";

    public T data;

    public static <T> Result<T> ok(T data){
        Result<T> result = new Result<>();
        result.data = data;
        return result;
    }

    public static Result<Object> error(String msg){
        Result<Object> result = new Result<>();
        result.code = -1;
        result.msg = msg;
        return result;
    }

    public static Result<Object> okMsg(String msg){
        Result<Object> result = new Result<>();
        result.msg = msg;
        return result;
    }
}

File: kt-sb-web/src/main/kotlin/org/example/entity/Result.kt
package org.example.entity

class Result<T>(
    var code: Int = 0,
    var msg: String = "成功",
    var data: T? = null
) {
    companion object {
        fun <T> ok(data: T?): Result<T> {
            return Result(data = data)
        }
        fun <T> fail(msg: String): Result<T> {
            return Result(-1, msg)
        }
    }
}
